notifier/manager_test.go

```diff
@@ -27,9 +27,11 @@ import (
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
+	prom_testutil "github.com/prometheus/client_golang/prometheus/testutil"
 	config_util "github.com/prometheus/common/config"
 	"github.com/prometheus/common/model"
 	"github.com/prometheus/common/promslog"
+	"github.com/stretchr/testify/assert" //nolint:depguard // assert.CollectT is used in require.EventuallyWithT()
```

Comment:
I'm afraid we cannot sneak this in.
I agree with you it could be useful/simplify the code, but it can also be misused.
let's avoid if we can do without, be cause we would only have once check (see below) we can do as we do in others `require.Eventually(t` in the code.
`return prom_testutil.ToFloat64(h.metrics.errors.WithLabelValues(serverX.URL)) > 0.0`